internal/cli/internal/command/services/postgres/launch.go

```diff
@@ -0,0 +1,308 @@
+			return err
+		}
+		region = r.Code
+	}
+
+	client := client.FromContext(ctx)
+
+	imageRef, err := client.API().GetLatestImageTag(ctx, "flyio/postgres")
+	if err != nil {
+		return err
+	}
+
+	config := PostgresProvisionConfig{
+		AppName:            name,
+		ConsulUrl:          consulUrl,
+		InitialClusterSize: initialClusterSize,
+		Password:           password,
+		Region:             region,
+		VolumeSize:         volumeSize,
+		VMSize:             vmSize,
+		SnapshotID:         snapshotID,
+		ImageRef:           imageRef,
+		Organization:       org,
+	}
+
+	pg := &Launch{
+		config: config,
+		client: client,
+	}
+
+	if err = pg.Launch(ctx); err != nil {
+		return fmt.Errorf("failed launching postgres: %w", err)
+	}
+
+	return nil
+}
+
+func (p *Launch) Launch(ctx context.Context) error {
+
+	app, err := p.createApp(ctx)
+	if err != nil {
+		return err
+	}
+
+	secrets, err := p.setSecrets(ctx)
+	if err != nil {
+		return err
+	}
+
+	io := iostreams.FromContext(ctx)
+	for i := 0; i < p.config.InitialClusterSize; i++ {
+		fmt.Fprintf(io.Out, "Provisioning %d of %d machines with image %s\n", i+1, p.config.InitialClusterSize, p.config.ImageRef)
+
+		machineConf, err := p.configurePostgres()
+		if err != nil {
+			return err
+		}
+
+		launchInput := api.LaunchMachineInput{
+			AppID:   app.ID,
+			OrgSlug: p.config.Organization.ID,
+			Region:  p.config.Region,
+			Config:  &machineConf,
+		}
+
+		machine, _, err := p.client.API().LaunchMachine(ctx, launchInput)
+		if err != nil {
+			return err
+		}
+
+		if err = waitForMachineState(ctx, p.client, p.config.AppName, machine.ID, "started"); err != nil {
+			return err
+		}
+	}
+
+	fmt.Fprintf(io.Out, "Provision complete\n\n")
+
+	connStr := fmt.Sprintf("postgres://postgres:%s@%s.internal:5432\n", secrets["OPERATOR_PASSWORD"], p.config.AppName)
+
+	fmt.Fprintf(io.Out, "Any app within the %s organization can connect to this Postgres using the following credentials:\n", p.config.Organization.Name)
+	fmt.Fprintf(io.Out, "  Username:    postgres\n")
+	fmt.Fprintf(io.Out, "  Password:    %s\n", secrets["OPERATOR_PASSWORD"])
+	fmt.Fprintf(io.Out, "  Hostname:    %s.internal\n", p.config.AppName)
+	fmt.Fprintf(io.Out, "  Proxy port:  5432\n")
+	fmt.Fprintf(io.Out, "  Postgres port:  5433\n")
+	fmt.Fprintf(io.Out, "  Connection string:  %s\n", connStr)
+	fmt.Fprintf(io.Out, "Save your credentials in a secure place, you won't be able to see them again!\n")
+
+	return nil
+}
+
+func (p *Launch) configurePostgres() (api.MachineConfig, error) {
+	machineConfig := flyctl.NewMachineConfig()
+
+	// Set env
+	env := map[string]string{
+		"PRIMARY_REGION": p.config.Region,
+	}
+	machineConfig.SetEnvVariables(env)
+
+	machineConfig.Config["size"] = p.config.VMSize
+	machineConfig.Config["image"] = p.config.ImageRef
+	machineConfig.Config["restart"] = map[string]string{
+		"policy": "no",
+	}
+
+	// Set mounts
+	var volumeHash string
+	var err error
+	if volumeHash, err = helpers.RandString(5); err != nil {
```

Comment:
you can just put `volumeHash, err := ...` on its own line here instead of using var twice.